OctopusDSC/DSCResources/cOctopusSeqLogger/cOctopusSeqLogger.psm1

```diff
@@ -0,0 +1,440 @@
+    throw "Property 'SeqServer' should be supplied if 'Ensure' is set to 'Present'"
+  }
+
+  Get-TargetResource -InstanceType $InstanceType `
+                     -Ensure $Ensure `
+                     -SeqServer $SeqServer `
+                     -SeqApiKey $SeqApiKey `
+                     -Properties $null
+
+  if ($InstanceType -eq "Tentacle") {
+    $dllPath = "$($env:ProgramFiles)\Octopus Deploy\Tentacle\Seq.Client.NLog.dll"
+  } elseif ($InstanceType -eq "OctopusServer") {
+    $dllPath = "$($env:ProgramFiles)\Octopus Deploy\Octopus\Seq.Client.NLog.dll"
+  }
+
+  if ($InstanceType -eq "Tentacle") {
+    $nlogConfigFile = "$tentacleExePath.nlog"
+  } elseif ($InstanceType -eq "OctopusServer") {
+    $nlogConfigFile = "$octopusServerExePath.nlog"
+  }
+
+  if ($Ensure -eq "Absent") {
+    if (Test-Path $dllPath) {
+      try {
+        Remove-Item $dllPath -Force
+      }
+      catch {
+        Write-Verbose "We tried to removing the seq dll from $dllPath"
+        Write-Verbose "But, we couldn't actually remove it, as its locked by Octopus.Server.exe / Tentacle.exe"
+      }
+    }
+    if (Test-Path $nlogConfigFile) {
+      Write-Verbose "Removing settings from $nlogConfigFile"
+      $nlogConfig = Get-NLogConfig $nlogConfigFile
+
+      $nlogExtensionElement = ($nlogConfig.nlog.extensions.add | where-object { $_.assembly -eq "Seq.Client.NLog" })
+      if ($null -ne $nlogExtensionElement) {
+        $nlogConfig.nlog.extensions.RemoveChild($nlogExtensionElement)
+      }
+      $nlogTargetElement = ($nlogConfig.nlog.targets.target | where-object {$_.name -eq "seq"})
+      if ($null -ne $nlogTargetElement) {
+        $nlogConfig.nlog.targets.RemoveChild($nlogTargetElement)
+      }
+      $nlogRuleElement = ($nlogConfig.nlog.rules.logger | where-object {$_.writeTo -eq "seq"})
+      if ($null -ne $nlogRuleElement) {
+        $nlogConfig.nlog.rules.RemoveChild($nlogRuleElement)
+      }
+      Write-Verbose "Saving updated config file $nlogConfigFile"
+      Save-NlogConfig $nlogConfig $nlogConfigFile
+    }
+  } else {
+     if (-not (Test-Path $dllPath)) {
+      Request-SeqClientNlogDll $dllPath
+    }
+    Write-Verbose "Modifying config file $nlogConfigFile"
+    $nlogConfig = Get-NLogConfig $nlogConfigFile
+
+    #remove then re-add "<add assembly="Seq.Client.NLog"/>" to //nlog/extensions
+    $nlogExtensionElement = ($nlogConfig.nlog.extensions.add | where-object { $_.assembly -eq "Seq.Client.Nlog" })
+    if ($null -ne $nlogPropertyElement) {
+      $nlogConfig.nlog.extensions.RemoveChild($nlogExtensionElement)
+    }
+    $newChild = $nlogConfig.CreateElement("add", $nlogConfig.DocumentElement.NamespaceURI)
+    $newChild.Attributes.Append((New-XmlAttribute $nlogConfig "assembly" "Seq.Client.NLog"))
+    $nlogConfig.nlog.extensions.AppendChild($newChild)
+
+    #remove then re-add "
+    #  <target name="seq" xsi:type="Seq" serverUrl="https://seq.octopushq.com" apiKey="wGdZV8CUL92KsGqZNVDX">
```

Comment:
Is this a valid API key?

Reply:
Whoops. It was, yeah. Disabled on the seq side.